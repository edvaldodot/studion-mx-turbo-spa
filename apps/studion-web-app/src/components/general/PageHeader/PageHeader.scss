@import '../variables';

.page-header {
  display: flex;
  gap: 4px;
  height: 116px;
  margin: 32px;

  &__content {
    flex: 1;
  }
  &__indicator,
  &__section,
  &__description {
    font-family: var(--font-family);
    line-height: 150%;
  }
  &__indicator,
  &__description {
    font-size: 16px;
  }
  &__indicator {
    font-weight: 700;
    color: var(--primary-color);
  }
  &__section {
    font-weight: 900;
    font-size: 40px;
    color: var(--black-700);
  }
  &__description {
    font-weight: 400;
    color: var(--black-900);
  }
  &__actions {
    flex: 1;
    display: flex;
    justify-content: flex-end;
    align-items: center;
  }

  @media (max-width: $sm) {
    flex-direction: column;

    &__actions {
      flex: 0;
      justify-content: flex-start;
    }
  }

  @media (max-width: $md) {
    flex-direction: column;

    &__actions {
      flex: 0;
      justify-content: flex-start;
    }
  }
}


