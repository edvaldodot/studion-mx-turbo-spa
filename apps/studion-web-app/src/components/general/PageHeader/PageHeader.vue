<template>
  <div class="page-header">
    <div class="page-header__content">
      <div class="page-header__indicator">
        <slot name="indicator" />
      </div>
      <div class="page-header__section">
        <slot name="section" />
      </div>
      <div class="page-header__description">
        <slot name="description" />
      </div>
    </div>
    <div class="page-header__actions">
      <slot name="actions" />
    </div>
  </div>
</template>
<style lang="scss">
@import 'PageHeader.scss';
</style>
