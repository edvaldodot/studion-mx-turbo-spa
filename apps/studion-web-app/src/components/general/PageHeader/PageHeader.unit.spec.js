import { shallowMount } from '@vue/test-utils'
import PageHeader from './PageHeader.vue'

describe('PageHeader.vue', () => {
  it('renders slots correctly', () => {
    const wrapper = shallowMount(PageHeader, {
      slots: {
        indicator: '<div class="test-indicator">Indicator Content</div>',
        section: '<div class="test-section">Section Content</div>',
        description: '<div class="test-description">Description Content</div>',
        actions: '<div class="test-actions">Actions Content</div>',
      },
    })

    expect(wrapper.find('.test-indicator').exists()).toBe(true)
    expect(wrapper.find('.test-indicator').text()).toBe('Indicator Content')

    expect(wrapper.find('.test-section').exists()).toBe(true)
    expect(wrapper.find('.test-section').text()).toBe('Section Content')

    expect(wrapper.find('.test-description').exists()).toBe(true)
    expect(wrapper.find('.test-description').text()).toBe('Description Content')

    expect(wrapper.find('.test-actions').exists()).toBe(true)
    expect(wrapper.find('.test-actions').text()).toBe('Actions Content')
  })

  it('matches the snapshot', () => {
    const wrapper = shallowMount(PageHeader, {
      slots: {
        indicator: '<div>Indicator Content</div>',
        section: '<div>Section Content</div>',
        description: '<div>Description Content</div>',
        actions: '<div>Actions Content</div>',
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})

