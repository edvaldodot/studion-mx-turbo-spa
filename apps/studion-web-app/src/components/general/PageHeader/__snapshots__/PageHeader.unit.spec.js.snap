// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader.vue matches the snapshot 1`] = `
"<div class="page-header">
  <div class="page-header__content">
    <div class="page-header__indicator">
      <div>Indicator Content</div>
    </div>
    <div class="page-header__section">
      <div>Section Content</div>
    </div>
    <div class="page-header__description">
      <div>Description Content</div>
    </div>
  </div>
  <div class="page-header__actions">
    <div>Actions Content</div>
  </div>
</div>"
`;
